public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.test.ts

```diff
@@ -0,0 +1,325 @@
+import { interval, lastValueFrom, of } from 'rxjs';
+
+import { LogRowModel, MutableDataFrame, FieldType, LogLevel, dataFrameToJSON, DataQueryErrorType } from '@grafana/data';
+import { BackendDataSourceResponse } from '@grafana/runtime';
+
+import { genMockFrames, setupMockedLogsQueryRunner } from '../__mocks__/LogsQueryRunner';
+import { validLogsQuery } from '../__mocks__/queries';
+import { LogAction } from '../types';
+import * as rxjsUtils from '../utils/rxjs/increasingInterval';
+
+import { LOG_IDENTIFIER_INTERNAL, LOGSTREAM_IDENTIFIER_INTERNAL } from './CloudWatchLogsQueryRunner';
+
+describe('CloudWatchLogsQueryRunner', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  describe('describeLogGroup', () => {
+    it('replaces region correctly in the query', async () => {
+      const { runner, fetchMock } = setupMockedLogsQueryRunner();
+      await runner.describeLogGroups({ region: 'default' });
+      expect(fetchMock.mock.calls[0][0].data.queries[0].region).toBe('us-west-1');
+
+      await runner.describeLogGroups({ region: 'eu-east' });
+      expect(fetchMock.mock.calls[1][0].data.queries[0].region).toBe('eu-east');
+    });
+
+    it('should return log groups as an array of strings', async () => {
+      const data: BackendDataSourceResponse = {
+        results: {
+          A: {
+            frames: [
+              {
+                schema: {
+                  name: 'logGroups',
+                  refId: 'A',
+                  fields: [{ name: 'logGroupName', type: FieldType.string }],
+                },
+                data: {
+                  values: [
+                    [
+                      '/aws/containerinsights/dev303-workshop/application',
```

Comment:
would it make sense to simplify this to some a handful of fake values? I feel like 1-3 values would get the point across right? unless there's more going on here? In which case maybe we could break it up into more tests?

Reply:
Sounds reasonable to me, but I'm not sure. Feels like these tests can be improved a lot, but I've been trying to not touch them unless they are for example duplicating test fixtures and such. Is it ok if we tackle refactoring existing tests in another PR?